### PR TITLE
Adds a `parents` function

### PR DESCRIPTION
**Pull request by rofinn**
To make working with derived features easier.

### CODE REVIEW

File: src/FeatureDescriptors.jl [hunk 1]
```diff
@@ -54,6 +54,15 @@ function label(D::Type{<:Descriptor})
     return Symbol(s)
 end
 
+"""
+    parents(D::Descriptor) -> Vector
+
+Returns the parent [`Descriptor`](@ref) for each table names returned from
+[`sources`](@ref). Similar to `Base.parent`, we return input input `D` when there
```
Comment (glennmoy) [inline src/FeatureDescriptors.jl]:
```suggestion
[`sources`](@ref). Similar to `Base.parent`, we return input `D` when there
```

File: src/test_utils.jl [hunk 2]
```diff
@@ -22,6 +22,31 @@ FeatureDescriptors.sources(::Type{<:FakeDescriptor}) = ["fake_table"]
 FeatureDescriptors.quantity_key(::Type{<:FakeDescriptor}) = :quantity
 FeatureDescriptors.categorical_keys(::Type{<:FakeDescriptor}) = [:category1, :category2]
 
+
+"""
+    OtherFakeDescriptor <:Descriptor
```
Comment (glennmoy) [inline src/test_utils.jl]:
```suggestion
    OtherFakeDescriptor <: Descriptor
```

File: Project.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name = "FeatureDescriptors"
 uuid = "cb0561c3-42a1-46a3-8749-4ae46de0afcc"
 authors = ["Invenia Technical Computing Corporation"]
-version = "1.1.0"
+version = "1.2.0"
```
Comment (glennmoy) [inline Project.toml:4]: is this a breaking change? since it adds new requirements to the test interface?

Comment (rofinn) [reply Project.toml:4]: It does add a new requirement to the test interface, but our default behaviour for `parents` should mean that the new test passes by default. I'll run the internally dependencies using this change to confirm though.

Comment (rofinn) [reply Project.toml:4]: Okay, ran this on our internal codebase and the only issue I ran into is that the `union` in the test doesn't work if `sources`/`parents` is empty, so I've just added an extra condition.

File: src/FeatureDescriptors.jl [hunk 4]
```diff
@@ -54,6 +54,15 @@ function label(D::Type{<:Descriptor})
...
 end
 
+"""
+    parents(D::Descriptor) -> Vector
+
+Returns the parent [`Descriptor`](@ref) for each table names returned from
+[`sources`](@ref). Similar to `Base.parent`, we return input `D` when there
+is no parent for a source.
+"""
+parents(D::Type{<:Descriptor}) = fill(D, length(sources(D)))
```
Comment (glennmoy) [inline src/FeatureDescriptors.jl:64]: Is the idea that a descriptor is automatically its own parent unless specified otherwise?

Comment (rofinn) [reply src/FeatureDescriptors.jl:64]: That's correct. If we didn't want to do this then we should make this release breaking as the new interface tests will breaking downstream packages.

**Review by glennmoy — APPROVED** [review]
Assuming no other major changes are needed

**Comment by rofinn** [comment]
Yeah, this seems to be working as expected and I don't expect to be making significant revisions to the design anytime soon. I'll plan to merge it and make alterations to the design if they become necessary. As it stands, this isn't breaking, so it shouldn't be a huge commitment.

**Pull request merged**